src/profile/ProfileForm.tsx

```diff
@@ -37,585 +23,315 @@ interface ChangedUser {
+            type="text"
+            name="username"
+            disabled={props.user !== undefined}
+            validMessage={"This username is valid."}
+            isValid={
+              props.context.formState.dirtyFields.has("username") &&
+              !props.context.errors["username"]
+            }
+            ref={props.context.register({
+              maxLength: {
+                value: 150,
+                message: "USER.USERNAME.TOO_LONG",
+              },
+              required: true,
+              validate: async (username: string) => {
+                if (props.user !== undefined) {
+                  // if we're editing a user the field is disabled anyway, don't validate
+                  return;
                 }
-                isInvalid={
-                  state.errors.username && state.errors.username.length > 0
+                if (!(await validateUsernameAvailable(username))) {
+                  return Promise.resolve("USER.USERNAME.ALREADY_EXISTS");
                 }
-              />
-              <Form.Control.Feedback>
-                Username is valid and is available.
-              </Form.Control.Feedback>
-              <Form.Control.Feedback type="invalid">
-                <ul>
-                  {state.errors.username?.map((msg) => (
-                    <li key={msg}>{ERRORS[msg]}</li>
-                  ))}
-                </ul>
-              </Form.Control.Feedback>
-            </>
-          )}
+              },
+            })}
+          />
         </Col>
       </Form.Group>
-
-      <Form.Group as={Row} controlId="profileEmail">
+      <Form.Group as={Row}>
         <Form.Label column sm={2}>
-          Email
+          Name:
         </Form.Label>
-        <Col sm={10}>
-          <Form.Control
-            type="email"
-            name="email"
-            required
-            placeholder="example@example.com"
-            value={state.changedUser.email ?? user?.email ?? ""}
-            onChange={onChange}
-            isValid={(state.changedUser.email?.length ?? 0) > 0}
-            isInvalid={state.errors.email && state.errors.email.length > 0}
+        <Col sm={5}>
+          <Field
+            errors={props.context.errors}
+            type="text"
+            name="first_name"
+            placeholder="First Name"
+            ref={props.context.register({
+              maxLength: {
+                value: 30,
+                message: "USER.FIRST_NAME.TOO_LONG.30",
+              },
+              required: true,
+            })}
+          />
+        </Col>
+        <Col sm={5}>
+          <Field
+            errors={props.context.errors}
+            type="text"
+            name="last_name"
+            placeholder="Last Name"
+            ref={props.context.register({
+              maxLength: {
+                value: 150,
+                message: "USER.LAST_NAME.TOO_LONG.150",
+              },
+              required: true,
```

Comment:
Shouldn't be required; not all people have last names.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/53ece12a397300e594d32c8dc8ce2d7c9859a9ed